src/python/ensembl/io/genomio/genbank/extract_data.py

```diff
@@ -90,43 +91,38 @@ class FormattedFilesGenerator:
         "CDS",
     ]
 
-    def __init__(self, prod_name: str, gb_file: Path, prefix: str = ""):
+    def __init__(
+        self, prod_name: str, gb_file: PathLike, prefix: str, out_dir: Optional[PathLike] = None
+    ) -> None:
         self.prefix = prefix
         self.seq_records: List[SeqRecord] = []
         self.prod_name = prod_name
         self.gb_file = gb_file
-        self.files = GenomeFiles()
 
-    def set_prefix(self, prefix):
-        """
-        Define a prefix to add to the feature IDs
-        """
-        if prefix:
-            self.prefix = prefix
+        # Output the gff3 file
+        self.out_dir = self.set_output_dir(out_dir)
+        self.files = GenomeFiles(self.out_dir)
 
-    def set_production_name(self, prod_name):
+    def set_output_dir(self, out_dir: Optional[PathLike]) -> PathLike:
         """
-        Define a production_name for the genome
-        """
-        if prod_name:
-            self.prod_name = prod_name
+        Define out_dir to write the files in a desired location
 
-    def extract_gb(self, out_dir: Optional[PathLike]) -> Dict[str, Path]:
-        """Extract data from a Genbank file and create files from it."""
-        if out_dir is not None:
-            self.files = GenomeFiles(out_dir)
-        self.set_prefix(self.prefix)
-        self.set_production_name(self.prod_name)
-        self.parse_genbank(Path(self.gb_file))
+        Args:
+            out_dir: Location to write the files
+        """
+        if out_dir is None:
+            out_dir = Path.cwd()
+        out_dir = Path(out_dir)
 
-        # Output the gff3 file
-        return self.files
+        return out_dir
```

Comment:
It might be simpler to just pass Path(out_dir), as it would provide the current working dir if out_dir is not set, so something like:
```
self.files = GenomeFiles(Path(out_dir))
```
(No need to keep `self.out_dir` as well since it's not used anywhere else)